### ISSUE TITLE
withContentType should set response encoding

### ISSUE THREAD

**Issue by lukas-krecan** [jadler-mocking/jadler#64]
When using respond().withContentType("text/plain; charset=CP1250") I would expect that response encoding will be set automatically based on the content type.

**Comment by jandudek** [comment]
Right now the `withContentType` method is a shorthand version for `withHeader("ContentType", value)` and therefore doesn't parse the value at all. If you want to specify the encoding explicitly (and not use the UTF-8 default), you have to call the `withEncoding` method.

This allows to simulate a faulty server which claims the response is in a specific encoding (using the `Content-Type` header) however the body is in fact encoded using a different encoding. It is an edge case, definitely, but it may be handy.

Do you have any suggestions? Should Jadler support this edge case? Any ideas to make this more user friendly?

**Comment by Kreinoee** [comment]
I agree with @jandudek. Som servers just behave badle, and id should be possibly to stub out such a server with jadler. 

So if this should be implemented, it should somehow be made optional. This could be done by a making a overloaded version taking an ekstra boolean flag, or mayby an EnumSet if more switches like this is expected over time:
```
respond().withContentType(
    "text/plain; charset=CP1250", 
    EnumSet.of(withContentTypeSwitches.AUTO_SET_ENCODING))
```
